src/metabase/driver/mysql.clj

```diff
@@ -876,11 +876,12 @@
   ;; for the mysql database), so we can't query the full privileges of the current user.
   (when-not (mariadb? database)
     (let [conn-spec   (sql-jdbc.conn/db->pooled-connection-spec database)
+          db-name     (get-in database [:details :dbname])
           table-names (->> (jdbc/query conn-spec "SHOW TABLES" {:as-arrays? true})
                            (drop 1)
                            (map first))]
       (for [[table-name privileges] (table-names->privileges (privilege-grants-for-user conn-spec "CURRENT_USER()")
-                                                             (:name database)
+                                                             db-name
```

Comment:
@calherries, This was a bug that we didn't catch in our BE tests.
we should use the db from details, not the display name.  I only catch it because the e2e tests use a different display name on its test database.

see my change in test/metabase/driver/mysql_test.clj to make sure we covered this in tests.